Gems/Atom/Tools/AtomToolsFramework/Code/Source/DynamicNode/DynamicNode.cpp

```diff
@@ -72,6 +59,10 @@ namespace AtomToolsFramework
 
     void DynamicNode::RegisterSlots()
     {
+        m_config = {};
+        AtomToolsFramework::DynamicNodeManagerRequestBus::EventResult(
+            m_config, m_toolId, &AtomToolsFramework::DynamicNodeManagerRequestBus::Events::GetConfig, m_configId);
+
```

Comment:
It looks like RegisterSlots are only called by the constructor, which would be invoked before de-serialization occurs, so how does refresh the data using the de serialized data.

Reply:
RegisterSlots is called by a base class post load setup function that needs to be manually invoked after loading, pasting, duplicating, undo, redo. Since it's virtual, it probably should not be called from the constructor anyway. We can explore replacing this with some kind of builder pattern or using a different method for storing node and slot IDs and references.

https://github.com/o3de/o3de/blob/development/Gems/GraphModel/Code/Source/Model/Node.cpp#L59-L81